### ISSUE TITLE
Rails integration does not pass down `request_queuing` down to rack integration

### ISSUE THREAD

**Issue by dobrite** [DataDog/dd-trace-rb#520]
I am trying to enable [`request_queuing`](https://github.com/DataDog/dd-trace-rb/blob/master/docs/GettingStarted.md#http-request-queuing) tracking for a rails app.

It looks like `use :rails` enables the rack integration and passes some rails config through to rack [here](https://github.com/DataDog/dd-trace-rb/blob/master/lib/ddtrace/contrib/rails/framework.rb#L46-L55), but `request_queuing` is missing.

Is there a way to enable `request_queuing` using Rails. It is unclear if also including `use :rack, request_queuing: true` will merge the config setup by `rails`, or overwrite it.

Thanks!
Dave

**Comment by delner** [comment]
Hey @dobrite . We don't have a means of passing through the Rack configuration via Rails right now (although we are in the process of reworking our configuration to support exactly this), however, you should be able to do `use :rack`. It shouldn't overwrite configuration provided by Rails, only change the options you supply explicitly. Just put it after `use :rails` and it should be fine.

**Comment by delner** [comment]
@dobrite just wanted to follow up: did this work for you?

**Comment by dobrite** [comment]
It did, however request queueing wasn't grouped in with the rails service, but rather a separate service. I started down the path of trying to unify them, but now I'm on another project.

I'm assuming the above is more of a feature request than a bug so I'll close this as complete.

Thank you for the follow up!

**Comment by liamdawson** [comment]
Is `use :rack` still the canonical way to achieve this?

**Comment by delner** [comment]
Currently, yes.

**Comment by nateberkopec** [comment]
I think the docs could be clearer about the expected ordering. Currently there's nothing in the integration docs saying that you need to put the `rack` section after the `rails` section.

**Comment by delner** [comment]
@nateberkopec The plan we laid out before was to address this by putting the `rack` section nested within `rails`, as the `rails` integration is a meta-integration that configures and activates a set of other integrations at once.

The idea was something like:

```ruby
Datadog.configure do |c|
  c.use rails do |rails|
    rails.rack request_queuing: true
  end
end
```

Not implemented yet, but should hopefully ordering obsolete.

**Comment by jeremy** [comment]
Is there a "blessed" workaround for this on the 1.0 release?

**Comment by marcotc** [comment]
Today we override these 3 `rack` configurations when one configures `rails`: https://github.com/DataDog/dd-trace-rb/blob/08c823edb60f61f75c1d96876d599bbf47dd6c62/lib/datadog/tracing/contrib/rails/framework.rb#L69-L71

Any `rack` options not present there remain as they were before `c.tracing.instrument :rails` was invoked.

I think the best solution here would be to only override `rack` configuration during the `rails` configuration if those `rack` settings were not modified. This would allow for any ordering between `rails` and it's related integrations (not only `rack`, but we do the same configuration override for ActiveRecord, ActionCable, etc.).

This suggestion is still not implemented, so today the recommendation is to have `rails` configured first, then configure everything else.

**Comment by jeremy** [comment]
Fixed by #2082.
